crypto/store/loader_file.c

```diff
@@ -1507,6 +1507,7 @@ static OSSL_STORE_INFO *file_load(OSSL_STORE_LOADER_CTX *ctx,
             unsigned char *data = NULL; /* DER encoded data */
             long len = 0;               /* DER encoded data length */
 
+            OSSL_STORE_INFO_free(result); /* from any previous iteration */
```

Comment:
I suppose that result basically cannot be non-NULL here because otherwise the matchcount would be !=0 and we would not loop. @levitte ?

Reply:
I also thought about this and that's why I wrote "potential" on this part.
Yet maybe the `file_load()` function should not rely too much on well-behavior on the functions it calls in between?
At least the inserted `OSSL_STORE_INFO_free()` call does not hurt.
I do not have a witness for an actual mem leak here.

So we could also simply leave out this part of the PR in the interest of a quick inclusion of the other (certain) leak fixes.

Reply:
I would then also add `result = NULL;` after the free. Because without it it is also not completely clear a doublefree could not happen.

Reply:
Another option would be to add `result == NULL `as the condition for looping.

Reply:
> Another option would be to add `result == NULL `as the condition for looping.

I consider this solution most elegant - done.

Reply:
@t8m, is this fine now?